src/dotnet-structuring/MainWindow.xaml.cs

```diff
@@ -0,0 +1,169 @@
+        public int ProccessAmount { get; private set; }
+        public string Options { get; private set; }
+        public string OutputDirectory { get; private set; }
+        public string NETCommand { get; private set; }
+        public string ProjectName { get; private set; }
+        public List<string> Directories = new List<string>();
+
+        public MainWindow()
+        {
+            InitializeComponent();
+        }
+
+        private Templates templates = new Templates();
+        private int LogNum;
+        private string CurrentLog;
+
+        private void WireEventHandlers(RunStructuring e)
+        {
+            StructuringHandler handler = new StructuringHandler(OnIncommingEventLog);
+            e.LogEvent += handler;
+        }
+
+        public void OnIncommingEventLog(object sender, EventLogger e)
+        {
+            LogNum++;
+
+            this.Dispatcher.Invoke(() =>
+            {
+                CurrentLog = e.logs;
+            });
+            this.Dispatcher.Invoke(() => OutputBox.Text += (CurrentLog + Environment.NewLine));
+            this.Dispatcher.Invoke(() => OutputBox.Text = Regex.Replace(OutputBox.Text, @"[\r\n]{2,}", Environment.NewLine));
+            if (LogNum < ProccessAmount)
+            {
+                pbStatus.Value++;
+            }
+            if (CurrentLog == "Done.")
+            {
+                this.Dispatcher.Invoke(() => pbStatus.IsIndeterminate = true);
+                this.Dispatcher.Invoke(() => pbStatus.Value = 1);
+                Style style = this.Dispatcher.Invoke(() => this.FindResource("ProgressBarSuccess") as Style);
+                this.Dispatcher.Invoke(() => pbStatus.Style = style);
+            }
+        }
+
+        public async void ExecButton_Click(object sender, RoutedEventArgs e)
+        {
+            OutputBox.Text = "";
+            RunStructuring OutputLogs = new RunStructuring();
+            this.Dispatcher.Invoke(() => pbStatus.IsIndeterminate = true);
+            Style style = this.FindResource("ProgressBarWarningStripe") as Style;
+            this.Dispatcher.Invoke(() => pbStatus.Style = style);
+            WireEventHandlers(OutputLogs);
+            await OutputLogs.CreateScript(OutputDirectory, Directories, NETCommand, ProjectName);
+        }
+
+        public void ProjectTypeSelector_DropDownClosed(object sender, EventArgs e)
+        {
+            ProjectType = ProjectTypeSelector.Text;
+            if (ProjectType == "New Project")
+            {
+                TemplateSelector.IsEnabled = true;
+                ArtifactsCheckBox.IsEnabled = true;
+                BuildCheckBox.IsEnabled = true;
+                DocsCheckBox.IsEnabled = true;
+                LibCheckBox.IsEnabled = true;
+                PackagesCheckBox.IsEnabled = true;
+                SamplesCheckBox.IsEnabled = true;
+                TestCheckBox.IsEnabled = true;
+                DotNetNewOptionsBox.IsEnabled = true;
```

Comment:
pack das in eine Methode die einen Boolean annimmt. Dann sparst du dir die zwei langen IFs